pkg/action/install.go

```diff
@@ -736,21 +749,22 @@ OUTER:
 func (c *ChartPathOptions) LocateChart(name string, settings *cli.EnvSettings) (string, error) {
 	name = strings.TrimSpace(name)
 	version := strings.TrimSpace(c.Version)
-
-	if _, err := os.Stat(name); err == nil {
-		abs, err := filepath.Abs(name)
-		if err != nil {
-			return abs, err
-		}
-		if c.Verify {
-			if _, err := downloader.VerifyChart(abs, c.Keyring); err != nil {
-				return "", err
+	if !c.UseInMemoryCache {
+		if _, err := os.Stat(name); err == nil {
+			abs, err := filepath.Abs(name)
+			if err != nil {
+				return abs, err
+			}
+			if c.Verify {
+				if _, err := downloader.VerifyChart(abs, c.Keyring); err != nil {
+					return "", err
+				}
 			}
+			return abs, nil
+		}
+		if _, err := os.Stat(name); err == nil && (filepath.IsAbs(name) || strings.HasPrefix(name, ".")) {
```

Comment:
I don't understand this code.  If we get here, we know that the if statement had `os.Stat(name)` return an error.  So why are we checking it in this if statement?

Reply:
You are correct, this augmentation of if statement is unnecessary as os.Stat is being checked before as well.